auth/auth-store/src/main/java/com/bazaarvoice/emodb/auth/identity/TableAuthIdentityManager.java

```diff
@@ -80,12 +82,23 @@ private T convertDataStoreEntryToIdentity(String id, Map<String, Object> map) {
             return null;
         }
 
+        // Make a copy of the map to avoid mutating the method parameter as a side-effect
+        Map<String, Object> identityMap = Maps.newHashMap(map);
+
+        // Identities have been in use since before internal IDs were introduced.  To grandfather in those keys we'll
+        // use the hash of the identity's ID as the internal ID.
+        if (!identityMap.containsKey(INTERNAL_ID)) {
+            identityMap.put(INTERNAL_ID, Intrinsic.getId(map));
```

Comment:
Yes, this is always called with the map returned by a `dataStore` get or scan, so the map will include all intrinsics including "~id".